src/crypto/openssl/cose_sign.cpp

```diff
@@ -83,9 +79,83 @@ namespace
 
 namespace ccf::crypto
 {
+  std::optional<int> key_to_cose_alg_id(ccf::crypto::PublicKey_OpenSSL& key)
+  {
+    const auto cid = key.get_curve_id();
+    switch (cid)
+    {
+      case ccf::crypto::CurveID::SECP256K1:
+      case ccf::crypto::CurveID::SECP256R1:
+        std::cout << "Return ES256" << std::endl;
+        return T_COSE_ALGORITHM_ES256;
+      case ccf::crypto::CurveID::SECP384R1:
+        std::cout << "Return ES384" << std::endl;
+        return T_COSE_ALGORITHM_ES384;
+      default:
+        return std::nullopt;
+    }
+  }
+
+  COSEParametersFactory cose_params_int_int(int64_t key, int64_t value)
+  {
+    const size_t args_size = sizeof(key) + sizeof(value);
+    return COSEParametersFactory(
+      [=](QCBOREncodeContext* ctx) {
+        QCBOREncode_AddInt64ToMapN(ctx, key, value);
+      },
+      args_size);
```

Comment:
Is this right? There is surely some overhead for tagging?

Reply:
I've put `300` bytes as initial space to fit all the metadata on the `sign` side.

However, I was paranoid about submitting really big strings or byte sequences and hitting implicit memory access issues as a result, so I'm returning this params size to adjust the buffer estimate accordingly each time the calling side wants more headers to be passed.

Reply:
Talked offline.

Seems like adding a map pair is Tag + value + Tag + value.

For integers, 1 byte extra for `uint8_t  uMajorType;`
For strings/byte-arrays, 1+8 bytes extra `type + size`

Reply:
Done